src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button, Input } from '../../Global';
+import { createPuzzleInstance, getAllPuzzles } from '../../../services';
+
+const PuzzleGenerate = () => {
+  const [puzzleList, setPuzzleList] = useState([]);
+  const [hint, setHint] = useState('');
+  const [latitude, setLatitude] = useState('');
+  const [longitude, setLongitude] = useState('');
+  const [address, setAddress] = useState('');
+  const [puzzleId, setPuzzleId] = useState('');
+
+  const handleSubmit = () => {
+    console.log(hint, latitude, longitude, address, puzzleId);
+    createPuzzleInstance(
+      parseInt(puzzleId),
+      parseInt(longitude),
+      parseInt(latitude),
+      address,
+      hint
+    ).then(puzzleInstance => console.log(puzzleInstance));
+  };
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
+      const puzzles = await getAllPuzzles();
+      setPuzzleList(puzzles);
+      setPuzzleId(puzzles[0].id);
+    };
+    fetchPuzzles();
```

Comment:
this doesn't need to be a function if you're going to only call it in this effect and immediately after

Reply:
I needed to resolve the promise of getAllPuzzles() using await

Reply:
Could't this be done using `.then()` instead?

Reply:
it could but I just want to use async await, should I change?